substra/sdk/client.py

```diff
@@ -78,19 +221,81 @@ class Client:
 
     def __init__(
         self,
+        *,
+        configuration_name: Optional[str] = None,
```

Comment:
@SdgJlbl Why naming it `configuration_name` and not `name`? or `client_name`? 🤔

Reply:
That is the conclusion of the discussion we had after the daily yesterday. This name is only used for the configuration of the `Client`, and we don't want it to be confused with the organisation name (that you can get from `client.organization_info()`). 
We cannot use the organisation name, that is a characteristic of the backend, because we need to be able to connect to the backend to have access to it (chicken and egg problem). 
And we cannot set the organisation name to this value, because it's set in the backend when the backend is created (so earlier than the creation of the client). 
So we thought that an appropriate option would to be making very clear that this name is only used for configuring the client, and not stored afterwards to avoid the confusion of having two similar but not synchronised concepts.

Reply:
Thanks for the detailed explanation. This makes sense and rules out the `name` naming that could be confusing with the organisation name.

I don't have a strong opinion on this, but I think `client_name`, really dissociates from `organization_name` and is less confusing / new than `configuration_name`. As there might be confusion between `configuration_name` and `configuration_file`.